### ISSUE TITLE
Node-RED instance keeps crashing and restarting

### ISSUE THREAD

**Issue by cirosilvano** [flowforge/flowforge#2202]
### Current Behavior

Node-RED instance keeps crashing and restarting. It stays in the "starting" state forever.

![Screenshot 2023-05-31 111534](https://github.com/flowforge/flowforge/assets/24639264/1f6e4e5e-963a-40da-bdda-c3334fbfc423)

### Expected Behavior

Node-RED instance should be created and stay operative.

### Steps To Reproduce

1. Install FlowForge on Windows 11 following the official guide, using default configurations only.
2. Create a user and login
3. Create an application in the starting page

### Environment

- FlowForge version: 1.7.1
- Node.js version: 18.15.0
- npm version: 9.5.0
- Platform/OS: Windows 11
- Browser: Chrome

### Have you provided an initial effort estimate for this issue?

I am not a FlowForge team member

**Comment by hardillb** [comment]
Just checking, this is the localfs build installed from [here](github.com/flowforge/installer)?

This is something we have seen when using the Docker build if the DNS settings are incorrect, but that should not come into play with a localfs build as everything should be using `localhost`

Have you made any changes to the `etc\flowforge.yml` file that ships with the install?

Assuming you've installed FlowForge into `C:\flowforge` and I ask you to do the following to try and gather some more information:

1. edit `C:\flowforge\app\node_modules\@flowforge\nr-launcher\lib\launcher.js`. Starting at line 345 add the following 2 `console.log` lines
```
                }
                // Use a HEAD request to minimise data transfer
                console.log(`liveness check URL - ${this.settings.baseURL}`) // add this line
                const res = await got.head(this.settings.baseURL, opts)
                statusCode = res.statusCode || 500
            } catch (error) {
                console.log(`liveness check error - ${error}`) // and this line
                statusCode = error.response?.statusCode || 500
            }
```
2. Then delete the instance and recreate it (you should not need to restart the whole FlowForge app)
3. Watch the logs for the instance until it has shown the same `Node-RED hang detected` messages
4. Look in the `C:\flowforge\var\projects` directory, in there there will be a directory that's name is a UUID (long string of numbers and letters). In that directory there will be a file called `out.log`.
5. Please provide the `out.log` file as an attachment here.

**Comment by cirosilvano** [comment]
Thanks for your reply. Yes, it is the flowforge-installer.zip file downloaded from the GitHub page, linked from the [FlowForge local installation guide](https://flowforge.com/docs/install/local/).

No change at all has been made to any file within the installation folder, which has been left as completely stock.

I'm leaving the log file as attachment here.
[out.log](https://github.com/flowforge/flowforge/files/11611914/out.log)

**Comment by hardillb** [comment]
Thanks for the log file. This points to what the problem is.

NodeJS has started to prefer IPv6 addresses over IPv4 in newer versions of NodeJS and the Node-RED instance is only listening on IPv4.

I will let you know when we have a fix.

**Comment by cirosilvano** [comment]
Thanks for your clear response and availability. I'll try downgrading my local NodeJS version as a temporary fix.

**Comment by hardillb** [comment]
Fix should ship in 1.8.0 next week
